### DISCUSSION TITLE
Ryven is missing key datatypes

### DISCUSSION THREAD

**Discussion by Spaced-out-games** [leon-thomm/Ryven#109] [General]
Ryven is missing some key data types that should be provided out-of-the-box.

bool,bytearray,bytes,classmethod,dict,enumerate,function,filter,list,map,object,range,str,set,staticmethod,slice,super,tuple,type, and zip are not supported.

Obviously, for some of them, it makes sense, like function, object, classmethod, object, and staticmethod, but Ryven desperately needs implementation of at least str, bool, dict, list, and type. Bytearray would be a plus.

I know some of these are partially implemented, like str, but what really hinders it is things like str.capitalize. The builtins module needs to be built upon so that Ryven can do more than just add/compare numbers out-of-the-box.

**Comment by leon-thomm** [comment]
I'm not sure what you mean exactly? Ryven does not care about datatypes, you can use any datatype you want... Maybe the `dtype` field of data inputs for nodes confused you? This is not actually saying anything about the expected datatype (yet), but only creates some default widget for the input.

Or are you asking about nodes for stuff like built-in functions? This is intentionally not included, as you then are usually operating at a way too high level of detail already. I think this paradigm only makes sense if it provides suitable abstraction, but not for capitalizing strings using a node (of course, there can be use cases where this is very useful, but not in general).

**Reply by Spaced-out-games** [reply]
what im getting at is that the base functionality is far too limited. The capitalize function is an example. For example, if I wanted to write an algorithm that takes the dictionary {"x":40,"z": 12, "y": 22} and converts it to (40, 22, 12) Ryven cannot natively do that. I'm suggesting simple python support. Basically, anything you can do without importing anything should be possible natively.
There's little list support (get elem, set elem, no pop, maybe append), no dictionary support as far as I can tell

there's a lot of stuff that could be implemented into Ryven but isn't. Only five or so nodes are available when you first start it up - set, get, set passive...

I know there's operator support which partially nullifies my point, but I digress

I hope that what I'm saying isn't sounding too harsh; I'm just trying to provide feedback.

**Comment by Robsel** [comment]
But how do I send a dict to another node tho
